src/main/java/io/cryostat/net/TargetConnectionManager.java

```diff
@@ -99,7 +104,14 @@ public void onRemoval(
                                         logger.info(
                                                 "Removing cached connection for {}",
                                                 descriptor.getTargetId());
+                                        JMXConnectionClosed evt =
+                                                new JMXConnectionClosed(descriptor.getTargetId());
+                                        evt.begin();
                                         connection.close();
```

Comment:
`connection.close()` may still throw an (unchecked) exception here, so wrapping this in `try/catch/finally` to ensure the event is recorded would be best. The exception can simply be rethrown after it is caught and the failure flagged in the event.

Reply:
I don't think an exception is thrown judging from the `JFRConnection` implementation in `cryostat.core`:

https://github.com/cryostatio/cryostat-core/blob/main/src/main/java/io/cryostat/core/net/JFRConnection.java#L199

From what I can tell, the exception is handled and not thrown upon closing.

Reply:
That's an internal implementation detail of -core, which is true right now but could change in the future.

Reply:
I flagged the failure in the event but when I tried to throw an exception I got the error `Exception Exception is not compatible with throws clause in RemovalListener<ConnectionDescriptor,JFRConnection>.onRemoval(ConnectionDescriptor, JFRConnection, RemovalCause)`.

It seems you can't throw a checked exception if the original method being overridden doesn't throw an exception as well, but for our purposes simply flagging the failure should be enough, right?

Reply:
We don't want to modify any control flow or other behaviour while adding these events, so swallowing the exception and not re-throwing it is not desirable. If you change your catch clause to catch and rethrow `RuntimeException` instead of simply `Exception` then I think it will do what we need.